src/patchers/SwitchCasePatcher.js

```diff
@@ -0,0 +1,86 @@
+import NodePatcher from './NodePatcher.js';
+import type { Editor, Node, ParseContext, Token } from './types.js';
+
+export default class SwitchCasePatcher extends NodePatcher {
+  constructor(node: Node, context: ParseContext, editor: Editor, conditions: NodePatcher, consequent: NodePatcher) {
+    super(node, context, editor);
+    this.conditions = conditions;
+    this.consequent = consequent;
+    this.negated = false;
+  }
+
+  patchAsStatement() {
+    let { conditions, consequent } = this;
+    // `when a` → `a`
+    let whenToken = this.getWhenToken();
+    this.remove(whenToken.range[0], conditions[0].start);
+
+    this.getCommaTokens().forEach((comma) => {
+      this.remove(...comma.range);
+    });
+
+    conditions.forEach((condition) => {
+      condition.insertBefore('case ');
+      if (this.negated) {
```

Comment:
Yep, I think that's fine. I'm not aware of a situation where that could cause issues.

> On Feb 28, 2016, at 12:30 PM, Jeff Chan <notifications@github.com> wrote:
> 
> In src/patchers/SwitchCasePatcher.js:
> 
> > +    this.negated = false;
> > +  }
> > +
> > +  patchAsStatement() {
> > +    let { conditions, consequent } = this;
> > +    // `when a` → `a`
> > +    let whenToken = this.getWhenToken();
> > +    this.remove(whenToken.range[0], conditions[0].start);
> > +
> > +    this.getCommaTokens().forEach((comma) => {
> > +      this.remove(...comma.range);
> > +    });
> > +
> > +    conditions.forEach((condition) => {
> > +      condition.insertBefore('case ');
> > +      if (this.negated) {
> I can do that. I did it this way to match the expected test output https://github.com/decaffeinate/decaffeinate/blob/2a185dcc8cec0a7a665b23d33212861ba7b18fc1/test/switch_test.js#L170-L186 (which is also the same as the CoffeeScript output). If we negate the condition, then our EqualityPatcher will flip signs. That's okay, right?
> 
> —
> Reply to this email directly or view it on GitHub.
>